xfel/merging/application/statistics/deltaccint.py

```diff
@@ -0,0 +1,190 @@
+    hkl_resolution_bins = self.params.statistics.hkl_resolution_bins
+
+    hkl_set = [hkl for hkl in list(set(filtered['miller_index_asymmetric'])) if hkl in hkl_resolution_bins]
+    n_hkl = len(hkl_set)
+    hkl_map = dict(zip(hkl_set, range(n_hkl)))
+
+    expt_map = filtered.experiment_identifiers()
+
+    # N expts (all ranks) x N hkl (this rank)
+    sums     = np.zeros((len(all_expt_ids), n_hkl), float)
+    n        = np.zeros((len(all_expt_ids), n_hkl), int)
+    merged   = np.zeros((len(all_expt_ids), n_hkl), float)
+    variance = np.zeros((len(all_expt_ids), n_hkl), float)
+
+    for refls in reflection_table_utils.get_next_hkl_reflection_table(reflections=filtered):
+      hkl = refls['miller_index_asymmetric'][0]
+      if hkl not in hkl_map: continue
+      hkl_idx = hkl_map[hkl]
+      intensity = flumpy.to_numpy(refls['intensity.sum.value'])
+
+      # set the sum and n for all expts for this hkl
+      sums[:,hkl_idx] = np.sum(intensity)
+      n   [:,hkl_idx] = len(refls)
+
+      # For each reflection, subtract it off the experiment it came from, leaving it for the rest of the experiments
+      for i in range(len(refls)):
+        expt_idx = all_expts_map[expt_map[refls['id'][i]]]
+        sums[expt_idx,hkl_idx] -= intensity[i]
+        n   [expt_idx,hkl_idx] -= 1
+      merged[:,hkl_idx] = sums[:,hkl_idx]/n[:,hkl_idx]
+
+      # compute variance for each hkl (less the intensity from each experiment)
+      diff_sq_ = np.zeros(len(all_expt_ids))
+      for expt_idx in range(len(all_expt_ids)):
+        mean_intensity = merged[expt_idx,hkl_idx]
+        diff_sq_[expt_idx] = np.sum((intensity-mean_intensity)**2)
+
+      for i in range(len(refls)):
+        expt_idx = all_expts_map[expt_map[refls['id'][i]]]
+        mean_intensity = merged[expt_idx,hkl_idx]
+        diff_sq_[expt_idx] -= (intensity[i] - mean_intensity)**2
+
+      variance[:,hkl_idx] = diff_sq_ / n[:,hkl_idx]
+
+    # N expts (all ranks) x N bins
+    n_bins = resolution_binner.n_bins_used()
+    all_i_sums   = np.zeros((len(all_expt_ids), n_bins), float) # sum of the averaged intensities
+    all_i_n      = np.zeros((len(all_expt_ids), n_bins), int)   # count of the averaged intensities
+    all_var_sums = np.zeros((len(all_expt_ids), n_bins), float) # sums of the variances of the intensities
+
+    # Sum up and reduce the bins
+    for hkl in hkl_set:
+      bin_idx = hkl_resolution_bins[hkl] - 1
+      all_i_sums  [:,bin_idx] += merged[:,hkl_map[hkl]]
+      all_i_n     [:,bin_idx] += 1
+      all_var_sums[:,bin_idx] += variance[:,hkl_map[hkl]]
+
+    total_i_sums   = comm.reduce(all_i_sums,   MPI.SUM, 0)
+    total_i_n      = comm.reduce(all_i_n,      MPI.SUM, 0)
+    total_var_sums = comm.reduce(all_var_sums, MPI.SUM, 0)
+
+    # Broadcast the average intensities
+    if self.mpi_helper.rank == 0:
+      total_i_average_ = total_i_sums / total_i_n
+    else:
+      total_i_average_ = None
+    total_i_average = comm.bcast(total_i_average_, 0)
```

Comment:
All of this can be simplified using and sped up (probably unnoticable?) using `comm.allreduce`:
```
total_i_sums   = comm.allreduce(all_i_sums, op=MPI.SUM)
total_i_n      = comm.allreduce(all_i_n,    op=MPI.SUM)
total_var_sums = comm.reduce(all_var_sums,  op=MPI.SUM)
total_i_average = total_i_sums / total_i_n
```